src/util/tokenize.jl

```diff
@@ -0,0 +1,67 @@
+# ========================
+# A Tokenizer for Strings
+# ========================
+
+# Ben J. Ward, 2016
+
+
+module Tokenize
+
+export
+    Tokenizer, TokenizerResult, tokenize
+
+
+"""
+Tokenizer type that is responsible for splitting a string into token according to a
+regex specification. The regex specification is stored in `dict`.
+
+**Fields:**
+
+* `dict`:      A Dictionary of String keys and Regex values.
+* `tokenizer`: A regex that is generated from the specification.
+"""
+type Tokenizer
+    dict::Dict{ASCIIString, Regex}
```

Comment:
`ASCIIString` is deprecated in v0.5. Consider to use `Compat.ASCIIString` or `String`.